python/ray/data/_internal/datasource/iceberg_datasink.py

```diff
@@ -43,14 +45,17 @@ def __init__(
             catalog_kwargs: Optional arguments to use when setting up the Iceberg
                 catalog
             snapshot_properties: custom properties write to snapshot when committing
-            to an iceberg table, e.g. {"commit_time": "2021-01-01T00:00:00Z"}
+                to an iceberg table, e.g. {"commit_time": "2021-01-01T00:00:00Z"}
+            overwrite_filter: A boolean expression in case of a partial overwrite,
+                default is None, which means append.
```

Comment:
Can you please elaborate more on what exactly this filter is and how it helps with overwriting?